static/js/market.js

```diff
@@ -62,12 +62,13 @@ const market = async () => {
         if (this.activeStall) {
           products = products.filter(p => p.stall_id == this.activeStall)
         }
-        if (!this.searchText || this.searchText.length < 2) return products
+        const searchText = this.searchText.toLowerCase()
+        if (!searchText || searchText.length < 2) return products
```

Comment:
if `searchText` can be `undefined`/`null` then move `const searchText = this.searchText.toLowerCase()` after the `if`